docs/FormatsAndConventions.md

```diff
@@ -138,6 +138,25 @@ e.g. `Error alert email has very long subject #5958`
 > to compensate for GitHub's lack of strong linking between the two.
 > Assume there is an invisible prefix in front of the PR title `Fixes issue : ...`
 
+## References to code elements
+
+Follow these conventions when referring to code elements e.g. when referring to a function name from a commit message.
+The objective is to be as concise as possible without causing ambiguities. Therefore, omit optional details when
+those details are not pertinent to the context.
+
+### Java
+
+* Variables: `package.class#variable` (optional: `package`)
+* Methods: `package.class#method(paramTypes):returnType` (optional: `package`, `paramTypes`, `:returnType`)
+
+e.g.
+* `Person#name`
+* `Person#getName()` 
+* `Person#getName():String` 
+* `Person#setName()` -- parameters omitted because only one `setName(String)` in the `Person` class
```

Comment:
While I understand the rationale behind enforcing compulsory parameters, I really think commit messages should be given a free pass on this, because it might make the commit message unnecessary verbose.

I don't think the reference would be ambiguous if it is expected of the reader to read the information in a contextual manner.

For example, in the case of commit message, it would be strange for the developer to use the knowledge from the current state of the project to understand what is going on in the commit. The developer should be expected to establish the context using the snapshot of the repository at that point in time.

However, readers are not expected to do the same when reading user/developer documentation (because it is meant to be a "live" document), so the omission of parameters should be forbidden in this case.

That was how I interpreted this statement that is added in this PR: `Therefore, omit optional details when those details are not pertinent to the context.`